streams/src/main/java/org/apache/kafka/streams/kstream/internals/metrics/Sensors.java

```diff
@@ -20,12 +20,19 @@
 import org.apache.kafka.common.metrics.Sensor;
 import org.apache.kafka.common.metrics.stats.Avg;
 import org.apache.kafka.common.metrics.stats.Max;
+import org.apache.kafka.common.metrics.stats.Rate;
+import org.apache.kafka.common.metrics.stats.Sum;
+import org.apache.kafka.common.metrics.stats.Total;
 import org.apache.kafka.streams.processor.internals.InternalProcessorContext;
 import org.apache.kafka.streams.processor.internals.metrics.StreamsMetricsImpl;
 
 import java.util.Map;
+import java.util.concurrent.TimeUnit;
 
 public class Sensors {
+
+    private static final String PROCESSOR_NODE_METRICS_GROUP = "stream-processor-node-metrics";
```

Comment:
might as well consolidate this repeated string.

Reply:
I am always confused by metric. What it the group name used for (ie, semantic meaning)?

Reply:
Sorry for my confusion. Let me rephrase my current understanding:
A "metric" is a single value that we track and report as key-value pair. A Sensor groups multiple metric together for ease of use -- each metric within a sensor gets globally unique name that is put together with many parts.

1.  We prefix metric names (the `xxx-metrics:` part) with the `clientId` that is different for each `StreamsThread`.
2. What is the purpose of `group-name` in the metric-name, ie, what metrics should use the same or different group name 
3. Each sensor has a name that is added to all metric names within the sensor (ie, the sensor name groups all it's contains metrics)
4 we also put additional tags to add more meta information (task-id, processor-id) if appropriate

Reply:
To clarify: there are two entities: the metrics registry which organizes the metrics, and there is metrics reporter which regularly pulls from the registry to report the metric values. 

Inside metrics registry there is `sensor` which as you understand is just a way of grouping metrics into meaningful clusters. The `SensorName` is just an id for distinguishing sensors in the metrics registry (i.e. you will see logic like `if this sensor as already been created in the registry skip this step). A metric name presented in `MetricName` which contains groupName, tags, etc is just a logical entity in the registry. How to represent the metric names is up to the metrics reporter (different reporters can definitely represent it differently). As for the sensor names, they should never be seen outside the registry as the metrics reporter never exposed them.

Reply:
Thanks @guozhangwang 

I was just checking, and we also define `"stream-processor-node-metrics"` in `ProcessorNode` -- should we unify both, to have one constant only?

Think, we can also simplify `ProcessorNode#createTaskAndNodeLatencyAndThroughputSensors` and remove the `group` parameter.

Reply:
Actually, someone filed a Jira last week reporting that a heap analysis identified this exact group name as responsible for Megabytes of heap space, so perhaps we should consolidate it into one constant!

Reply:
If this is an issue, maybe we should do a fix first (in it's own PR, so we can cherry-pick), and rebase/merge this PR later? Thought? Are you planning to do a PR for the reported issue?